test/ibmq/websocket/test_websocket_integration.py

```diff
@@ -26,45 +33,129 @@
 class TestWebsocketIntegration(QiskitTestCase):
     """Websocket integration tests."""
 
-    def setUp(self):
-        qr = QuantumRegister(1)
-        cr = ClassicalRegister(1)
-        self._qc1 = QuantumCircuit(qr, cr, name='qc1')
-        self._qc1.measure(qr[0], cr[0])
-
+    # pylint: disable=arguments-differ
     @requires_qe_access
     @requires_new_api_auth
-    def test_websockets_simulator(self, qe_token, qe_url):
-        """Test checking status of a job via websockets for a simulator."""
+    def setUp(self, qe_token, qe_url):
         ibmq_factory = IBMQFactory()
-        provider = ibmq_factory.enable_account(qe_token, qe_url)
-        backend = provider.get_backend(simulator=True)
+        self.provider = ibmq_factory.enable_account(qe_token, qe_url)
+        self.sim_backend = self.provider.get_backend(simulator=True)
 
-        qc = transpile(self._qc1, backend=backend)
-        qobj = assemble(qc, backend=backend)
+        # Create a circuit
+        qr = QuantumRegister(1)
+        cr = ClassicalRegister(1)
+        self.qc1 = QuantumCircuit(qr, cr, name='qc1')
+        self.qc1.measure(qr[0], cr[0])
 
-        job = backend.run(qobj)
-        # Manually disable the non-websocket pooling.
-        job._wait_for_final_status = None
-        result = job.result()
+        # Create and run a default Qobj using the simulator.
+        self.circuit = transpile(self.qc1, backend=self.sim_backend)
+        self.qobj = assemble(self.circuit, backend=self.sim_backend, shots=1)
+        self.job = self.sim_backend.run(self.qobj)
```

Comment:
Can you take this call out of `setUp` and have the tests that need a job against the simulator run it explicitly (in a function or with this line directly, etc)? Mostly in order to avoid running extra jobs during the tests, and in general as we tend to avoid cramming up `setUp` for readability and performance (don't worry about the rest of changes to this `setUp` for the moment).